modules/multitile_support/src/tile_map.xc

```diff
@@ -4,10 +4,170 @@
 #include "platform.h"
 
 extern "C" {
+
+
+#ifndef MINIMISE_MAIN_TILE_ARGS 
+
+
+
 void main_tile0(chanend c0, chanend c1, chanend c2, chanend c3);
 void main_tile1(chanend c0, chanend c1, chanend c2, chanend c3);
 void main_tile2(chanend c0, chanend c1, chanend c2, chanend c3);
 void main_tile3(chanend c0, chanend c1, chanend c2, chanend c3);
+
+
+#else // MINIMISE_MAIN_TILE_ARGS defined
+
+
+
+// -------------------- Tile 0 prototype ------------------------------
+#if ((PLATFORM_SUPPORTS_TILE_0 == 1) && (PLATFORM_USES_TILE_0 == 1))
+                      void main_tile0(
+  
+// c0  
+// channel c0  is loopback so it is never in the minimised argument set
+
+
+// c1
+#if ((PLATFORM_SUPPORTS_TILE_1 == 1) && (PLATFORM_USES_TILE_1 == 1))
+                      chanend c1
+#define ARGS_STARTED
+#endif
+
+
+// c2
+#if ((PLATFORM_SUPPORTS_TILE_2 == 1) && (PLATFORM_USES_TILE_2 == 1))
+#ifdef ARGS_STARTED
+                      ,  
+#endif
+                      chanend c2
+#define ARGS_STARTED
+#endif
+
+
+// c3
+#if ((PLATFORM_SUPPORTS_TILE_3 == 1) && (PLATFORM_USES_TILE_3 == 1))
```

Comment:
would be good to have the function prototypes in a header file where the C file can include it. This allows the user to get compile time (as opposed to link time) errors when the define it incorrectly.

Reply:
This is a great suggestion. I have updated the PR to include a new header file main_tile.h which provides this.